### PR TITLE
add validation for ssh keys

### LINKED ISSUE #469: Validate SSH public keys in webhook
All public keys which get supplied within the machine object must be validated within the webhook.

Validation should work by just calling with https://github.com/golang/crypto/blob/master/ssh/keys.go#L256

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Add validation for ssh keys.

Fixes #469

A alerting rule for existing machines will be added to Kubermatic

```release-note
Validate SSH Keys
```

### CODE REVIEW

File: pkg/admission/machines_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+				"some invalid key",
+			},
+			err: errors.New("invalid public key 'some invalid key': ssh: no key found"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := validatePublicKeys(test.keys)
+			if err != nil {
```
Comment (alvaroaleman) [inline pkg/admission/machines_test.go]: Nit: This can be simplified as `if err! =test.err`

File: pkg/admission/machines.go [hunk 2]
```diff
@@ -112,3 +118,15 @@ func (ad *admissionData) defaultAndValidateMachineSpec(spec *clusterv1alpha1.Mac
 
 	return nil
 }
+
+func validatePublicKeys(keys []string) error {
+	for _, s := range keys {
+		//_, err := ssh.ParsePublicKey([]byte(s))
```
Comment (alvaroaleman) [inline pkg/admission/machines.go]: Remainder?

File: pkg/admission/machines.go [hunk 3]
```diff
@@ -112,3 +118,15 @@ func (ad *admissionData) defaultAndValidateMachineSpec(spec *clusterv1alpha1.Mac
...
+		_, _, _, _, err := ssh.ParseAuthorizedKey([]byte(s))
+		if err != nil {
+			return fmt.Errorf("invalid public key '%s': %v", s, err)
```
Comment (alvaroaleman) [inline pkg/admission/machines.go]: Nit: Use `%q` instead of `'%s' `

Comment (mrIncompetent) [reply pkg/admission/machines.go]: Done with https://github.com/kubermatic/machine-controller/pull/497/commits/5e3e9a738530605e8e89d8c06d607719903919bb

File: pkg/admission/machines.go (hunk 2 above)
Comment (mrIncompetent) [reply pkg/admission/machines.go]: Removed with https://github.com/kubermatic/machine-controller/pull/497/commits/5e3e9a738530605e8e89d8c06d607719903919bb

File: pkg/admission/machines_test.go (hunk 1 above)
Comment (mrIncompetent) [reply pkg/admission/machines_test.go]: Done with https://github.com/kubermatic/machine-controller/pull/497/commits/5e3e9a738530605e8e89d8c06d607719903919bb

File: pkg/admission/machines_test.go [hunk 4]
```diff
@@ -58,19 +59,8 @@ func TestValidatePublicKeys(t *testing.T) {
...
-				}
-				if err.Error() != test.err.Error() {
-					t.Errorf("Got an error '%v' while expected it to be: %v", err, test.err)
-				}
-			} else {
-				if test.err != nil {
-					// We got an error but did not expected it
-					t.Errorf("Got no error while expected one to be returned: %v", test.err)
-				}
+			if fmt.Sprint(err) != fmt.Sprint(test.err) {
```
Comment (alvaroaleman) [inline pkg/admission/machines_test.go]: Out of curiosity why do you do this instead of comparing `err.Error()`?

/lgtm
/approve

Comment (mrIncompetent) [reply pkg/admission/machines_test.go]: If err is nil, calling `err.Error()` would cause a NPE

**Pull request merged**